src/graphql/resolvers/ssb.js

```diff
@@ -42,6 +123,110 @@ module.exports = function Resolvers () {
     })
   }
 
+  /**
+   * Gets all the votes on a message
+   * TODO: doesnt handle unlikes and duplicates yet.
+   * @param {string} id - id of a message
+   */
+  const getVotes = (id) => {
+    return new Promise((resolve, reject) => {
+      pull(
+        ssb.db.query(
+          where(
+            votesFor(id)
+          ),
+          descending(), // latest => oldest
+          toPullStream()
+        ),
+        pull.map(msg => {
+          const vote = msg?.value?.content?.vote
+          return {
+            author: msg.value.author,
+            timestamp: msg.value.timestamp,
+            value: vote?.value,
+            expression: vote?.expression
+          }
+        }),
+        pull.collect((err, votes) => {
+          if (err) reject(err)
+          else resolve(votes)
+        })
+      )
+    })
+  }
+
+  /**
+   * Gets all the threads initiated by a certain profile id 
+   * @param {string} id - id of a profile
+   * @param {object} opts - optional parameters
+   * @param {int} [opts.limit=10] - max amount of threads to return
+   * @param {int} opts.threadMaxSize - max amount of messages in each thread to return
+   */
+  const getThreads = (id, opts) => {
+    const { threadMaxSize, limit = 10 } = opts
+    return new Promise((resolve, reject) => {
+      pull(
+        ssb.threads.profile({ id, reverse: true, threadMaxSize }),
+        pull.take(limit),
+        pull.collect((err, threads) => {
+          if (err) return reject(err)
+
+          const res = threads.map(({ messages }) => {
+            return {
+              id: messages[0].key,
+              messages
+            }
+          })
+
+          resolve(res)
+        })
+      )
+    })
+  }
+
+  /**
+   * Takes the messages from a thread and maps them based on whether
+   * a profile is returned for the author of that message or not.
+   * Messages from someone who hasnt yet opted in to publicWebHosting or
+   * a profile wasnt found for them, will return empty values
+   * @param {array} messages - messages in a thread
+   */
+  const mapMessages = (messages) => {
+    return new Promise((resolve, reject) => {
+      pull(
+        pull.values(messages),
+        pullParaMap((msg, cb) => {
+          getProfile(msg.value.author)
+            .then(profile => {
+              // if there was a profile, thats great
+              if (profile) {
+                return cb(null, {
+                  id: msg.key,
+                  author: msg.value.author,
+                  text: (msg.value.content.text || ''),
```

Comment:
Actually can you console.log the cases where `!msg.value.content.text` we've been seeing a few of those come through strangely